src/test/librust/sapling_wallet_tests.cpp

```diff
@@ -0,0 +1,1133 @@
+    builder.AddTransparentInput(COutPoint(), scriptPubKey, 50000);
+    builder.AddSaplingOutput(extfvk.fvk.ovk, pk, 40000, {});
+    auto tx1 = builder.Build().GetTxOrThrow();
+
+    BOOST_CHECK_EQUAL(tx1.vin.size(), 1);
+    BOOST_CHECK_EQUAL(tx1.vout.size(), 0);
+    BOOST_CHECK_EQUAL(tx1.sapData->vShieldedSpend.size(), 0);
+    BOOST_CHECK_EQUAL(tx1.sapData->vShieldedOutput.size(), 1);
+    BOOST_CHECK_EQUAL(tx1.sapData->valueBalance, -40000);
+
+    CWalletTx wtx {&wallet, tx1};
+
+    // Fake-mine the transaction
+    BOOST_CHECK_EQUAL(0, chainActive.Height());
+    SaplingMerkleTree saplingTree;
+    CBlock block;
+    block.vtx.emplace_back(MakeTransactionRef(wtx));
+    block.hashMerkleRoot = BlockMerkleRoot(block);
+    auto blockHash = block.GetHash();
+    CBlockIndex fakeIndex {block};
+    BlockMap::iterator mi = mapBlockIndex.emplace(blockHash, &fakeIndex).first;
+    fakeIndex.phashBlock = &((*mi).first);
+    chainActive.SetTip(&fakeIndex);
+    BOOST_CHECK(chainActive.Contains(&fakeIndex));
+    BOOST_CHECK_EQUAL(0, chainActive.Height());
+
+    // Simulate SyncTransaction which calls AddToWalletIfInvolvingMe
+    auto saplingNoteData = wallet.GetSaplingScriptPubKeyMan()->FindMySaplingNotes(wtx).first;
+    BOOST_CHECK(saplingNoteData.size() > 0);
+    wtx.SetSaplingNoteData(saplingNoteData);
+    wtx.SetMerkleBranch(mapBlockIndex[blockHash], 0);
+    wallet.LoadToWallet(wtx);
+
+    // Simulate receiving new block and ChainTip signal
+    wallet.IncrementNoteWitnesses(&fakeIndex, &block, saplingTree);
+    wallet.GetSaplingScriptPubKeyMan()->UpdateSaplingNullifierNoteMapForBlock(&block);
+
+    // Retrieve the updated wtx from wallet
+    uint256 hash = wtx.GetHash();
+    wtx = wallet.mapWallet[hash];
+
+    // Prepare to spend the note that was just created
+    auto maybe_pt = libzcash::SaplingNotePlaintext::decrypt(
+            tx1.sapData->vShieldedOutput[0].encCiphertext, ivk, tx1.sapData->vShieldedOutput[0].ephemeralKey, tx1.sapData->vShieldedOutput[0].cmu);
+    BOOST_CHECK(static_cast<bool>(maybe_pt));
+    auto maybe_note = maybe_pt.get().note(ivk);
+    BOOST_CHECK(static_cast<bool>(maybe_note));
+    auto note = maybe_note.get();
+    auto anchor = saplingTree.root();
+    auto witness = saplingTree.witness();
+
+    // Create a Sapling-only transaction
+    // 0.0004 z-ZEC in, 0.00025 z-ZEC out, 0.0001 t-ZEC fee, 0.00005 z-ZEC change
+    auto builder2 = TransactionBuilder(consensusParams, 2);
+    builder2.AddSaplingSpend(expsk, note, anchor, witness);
+    builder2.AddSaplingOutput(extfvk.fvk.ovk, pk, 25000, {});
+    auto tx2 = builder2.Build().GetTxOrThrow();
+
+    BOOST_CHECK_EQUAL(tx2.vin.size(), 0);
+    BOOST_CHECK_EQUAL(tx2.vout.size(), 0);
+    BOOST_CHECK_EQUAL(tx2.sapData->vShieldedSpend.size(), 1);
+    BOOST_CHECK_EQUAL(tx2.sapData->vShieldedOutput.size(), 2);
+    BOOST_CHECK_EQUAL(tx2.sapData->valueBalance, 10000);
+
+    CWalletTx wtx2 {&wallet, tx2};
+    auto hash2 = wtx2.GetHash();
```

Comment:
(9cabc2ed2399221b23491c536254488dda44e60e)
It seems that  this variable is never used.